compositor_pipeline/src/pipeline/output/whip/init_peer_connection.rs

```diff
@@ -0,0 +1,183 @@
+use crate::{
+    audio_mixer::AudioChannels,
+    pipeline::{AudioCodec, VideoCodec},
+};
+
+use super::{WhipAudioOptions, WhipCtx, WhipError};
+use std::{
+    env::{self, VarError},
+    sync::Arc,
+};
+use tracing::{info, warn};
+use webrtc::{
+    api::{
+        interceptor_registry::register_default_interceptors,
+        media_engine::{MediaEngine, MIME_TYPE_H264, MIME_TYPE_OPUS},
+        APIBuilder,
+    },
+    ice_transport::ice_server::RTCIceServer,
+    interceptor::registry::Registry,
+    peer_connection::{configuration::RTCConfiguration, RTCPeerConnection},
+    rtp_transceiver::{
+        rtp_codec::{RTCRtpCodecCapability, RTCRtpCodecParameters, RTPCodecType},
+        rtp_transceiver_direction::RTCRtpTransceiverDirection,
+    },
+    track::track_local::track_local_static_rtp::TrackLocalStaticRTP,
+};
+
+const STUN_SERVER_ENV: &str = "LIVE_COMPOSITOR_STUN_SERVERS";
+
+pub async fn init_peer_connection(
+    whip_ctx: &WhipCtx,
+) -> Result<
+    (
+        Arc<RTCPeerConnection>,
+        Option<Arc<TrackLocalStaticRTP>>,
+        Option<Arc<TrackLocalStaticRTP>>,
+    ),
+    WhipError,
+> {
+    let mut media_engine = MediaEngine::default();
+    media_engine.register_default_codecs()?;
+
+    if let Some(video) = whip_ctx.options.video {
+        media_engine.register_codec(video_codec_parameters(video), RTPCodecType::Video)?;
+    }
+
+    if let Some(audio) = whip_ctx.options.audio {
+        media_engine.register_codec(
+            audio_codec_parameters(audio, whip_ctx.sample_rate)?,
+            RTPCodecType::Audio,
+        )?;
+    }
+    let mut registry = Registry::new();
+    registry = register_default_interceptors(registry, &mut media_engine)?;
+    let api = APIBuilder::new()
+        .with_media_engine(media_engine)
+        .with_interceptor_registry(registry)
+        .build();
+
+    let mut stun_servers_urls = vec!["stun:stun.l.google.com:19302".to_owned()];
+
+    match env::var(STUN_SERVER_ENV) {
```

Comment:
do not read env variable inside the create, everything should be read in config.rs in top-level create and passed as envs